### PR TITLE
Add additional tests for component installation

### PR DESCRIPTION
**Pull request by LiquidPT**
Adding some tests for components we install that aren't being checked now.

### CODE REVIEW

File: tests/test-definitions.sh [hunk 1]
```diff
@@ -99,9 +99,8 @@ function verify_cuda_installation {
     check_exit_code "NVIDIA Peer memory module is inserted" "NVIDIA Peer memory module is not inserted!"
 
     # Verify if CUDA is installed
-    # re-enable this after testing
```
Comment (yakovdyadkin) [inline tests/test-definitions.sh:102]: Why was this commented before?

Comment (LiquidPT) [reply tests/test-definitions.sh:102]: oh, I need to recomment that out. It works ion VMs when I test, but fails in the pipeline for some reason. says that nvcc isn't a known command

File: tests/test-definitions.sh [hunk 2]
```diff
@@ -210,6 +215,36 @@ function verify_ipoib_status {
...
+        * ) ;;
+    esac
+    check_exit_code "GDRCopy Installed" "GDRCopy not installed!"
+}
+
+function verify_pssh_installation {
+    # Verify PSSH package installation
+    case ${ID} in
+        ubuntu) dpkg -l | grep pssh;;
+        almalinux) dnf list installed | grep pssh;;
```
Comment (yakovdyadkin) [inline tests/test-definitions.sh:242]: I have seen pssh appear under different name such as `parallel-ssh` is the package name parallel ssh `pssh` across all distros?

**Comment by yakovdyadkin** [comment]
General comment, with only 3 unique distro + version combinations in _tests/test-matrix.json_ that are almost identical in content, does it make sense to move the common tests that each distro specifies to `verify_common_components` func in _tests/run-tests.sh_?

**Review by yakovdyadkin — COMMENTED** [review]
General comment, with only 3 unique distro + version combinations in tests/test-matrix.json that are almost identical in content, does it make sense to move the common tests that each distro specifies to verify_common_components func in tests/run-tests.sh?

**Review by moes1 — APPROVED** [review]

File: tests/test-definitions.sh (hunk 2 above)
Comment (LiquidPT) [reply tests/test-definitions.sh:242]: on both Alma and Ubuntu we install a package named "pssh". 

We seem to use a mix of parallel ssh packages on our validation pipeline and I don't know why.

**Comment by LiquidPT** [comment]
> General comment, with only 3 unique distro + version combinations in tests/test-matrix.json that are almost identical in content, does it make sense to move the common tests that each distro specifies to verify_common_components func in tests/run-tests.sh?

Maybe. There were certainly things that were different that I've recently made the same. I think I'd like to get Mariner in the mix before I make the call of "these things are also always installed".

**Pull request merged**